lib/resourceapply/security.go

```diff
@@ -29,14 +29,20 @@ func ApplySecurityContextConstraintsv1(ctx context.Context, client securityclien
 		return nil, false, nil
 	}
 
+	var original securityv1.SecurityContextConstraints
+	existing.DeepCopyInto(&original)
 	modified := pointer.Bool(false)
-	resourcemerge.EnsureObjectMeta(modified, &existing.ObjectMeta, required.ObjectMeta)
+	resourcemerge.EnsureSecurityContextConstraints(modified, existing, *required)
```

Comment:
`EnsureSecurityContextConstraints` will break when a new field is added into SCCs. Do we need any additional logic that makes this better than applying a generic unstructured object?

Reply:
As discussed with @vrutkovs, there might be some advantages to keeping this typed. In particular, Vadim named cases of slice order changes and merging map keys.
The SCCs unfortunately don't have `Spec` that would define their contents. I would suggest combining `DeepCopy` of the whole SCC into the final object while preserving the logic around `ObjectMeta`.

Reply:
The current treatment follows what CVO does even for other types so I'm not inclined to change that in a bugfix (consistency has value). What you say is right (xref e.g. https://github.com/openshift/cluster-version-operator/pull/965) but it will need to be fixed separately.